src/ruby/ext/grpc/rb_call.c

```diff
@@ -179,6 +179,36 @@ static VALUE grpc_rb_call_cancel(VALUE self) {
   return Qnil;
 }
 
+/* Not a part of the surface API.
+ * This is meant for internal usage by the "write thread" of grpc-ruby
+ * client-side bidi calls. It provides a way for the background write-thread
+ * to propogate failures to the main read-thread and give the user an error
+ * message. */
+static VALUE grpc_rb_call_cancel_with_bidi_write_thread_error_message(
```

Comment:
I think it would be simpler to just surface `cancel_with_status`, and call it in the Ruby code. And in the future, you might want to use `cancel_with_status` for a different error.